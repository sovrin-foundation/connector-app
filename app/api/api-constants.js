// @flow

export const TOKEN_EXPIRED_CODE = 'VES-101'
export const SERVER_ERROR_CODE = 'OCS-000'
export const SERVER_API_CALL_ERROR = 'OCS-001'
export const PENDING_CONNECTION_REQUEST_CODE = 'CRS-103'
export const PUSH_NOTIFICATION_SENT_CODE = 'push-notification-sent'

export const QR_CODE_REQUEST_ID = 'id'
export const QR_CODE_SENDER_DETAIL = 's'
export const QR_CODE_SENDER_DID = 'd'
export const QR_CODE_SENDER_NAME = 'n'
export const QR_CODE_LOGO_URL = 'l'
export const QR_CODE_SENDER_VERIFICATION_KEY = 'v'
export const QR_CODE_SENDER_KEY_DELEGATION = 'dp'
export const QR_CODE_DELEGATION_DID = 'd'
export const QR_CODE_DELEGATION_KEY = 'k'
export const QR_CODE_DELEGATION_SIGNATURE = 's'
export const QR_CODE_SENDER_AGENCY = 'sa'
export const QR_CODE_SENDER_AGENCY_DID = 'd'
export const QR_CODE_SENDER_AGENCY_KEY = 'v'
export const QR_CODE_SENDER_AGENCY_ENDPOINT = 'e'
export const QR_CODE_TARGET_NAME = 't'
export const GENESIS_FILE_NAME = 'pool_transactions_genesis'

export const MESSAGE_TYPE = {
  AUTH: 'auth-req',
  CLAIM_OFFER: 'credOffer',
  CLAIM_REQUEST: 'credReq',
  PROOF_REQUEST: 'proofReq',
  PROOF: 'proof',
  CLAIM: 'cred',
}

export const PAYLOAD_TYPE = {
  GET_MESSAGE: 'GET_MSGS',
  UPDATE_PUSH_COM_METHOD: 'UPDATE_PUSH_COM_METHOD',
  SEND_MESSAGE: 'SEND_MSG',
}

export const STATUS_CODE = {
  ACCEPT: 'MS-104',
}

export const ERROR_ALREADY_EXIST = {
  code: 'OCS',
  message: 'You already have a relationship with ',
}

export const ERROR_ALREADY_EXIST_TITLE = 'Relationship already established'

export const ERROR_INVITATION_RESPONSE_PARSE_CODE = 'RN-100'

export const ERROR_INVITATION_RESPONSE_PARSE =
  'Error parsing error message in invitation response'

export const ERROR_PENDING_INVITATION_RESPONSE_PARSE_CODE = 'RN-101'

export const ERROR_PENDING_INVITATION_RESPONSE_PARSE =
  'Error parsing error message Pending Invitation SMS'

export const ERROR_INVITATION_RESPONSE_FAILED =
  'Failed to establish connection. Please try again later.'
