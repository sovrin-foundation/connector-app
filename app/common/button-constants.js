// @flow
export const DENY = 'Deny'
export const CONNECT = 'Connect'
