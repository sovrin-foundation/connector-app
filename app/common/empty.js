// @flow
// sometimes we need an empty object to use as a placeholder in destructuring
export default {}
