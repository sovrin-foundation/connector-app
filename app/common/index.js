// @flow
export * from './button-constants'
export * from './message-constants'
export * from './route-constants'
export * from './secure-storage-constants'
export function noop() {}
