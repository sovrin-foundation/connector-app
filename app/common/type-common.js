// @flow

import type { Element } from 'react'
import type { NavigationScreenProp } from 'react-navigation'

export type CustomError = {
  code: string,
  message: string,
}

export const INITIAL_TEST_ACTION: 'INITIAL_TEST_ACTION' = 'INITIAL_TEST_ACTION'

export type InitialTestAction = {
  type: typeof INITIAL_TEST_ACTION,
}

export const initialTestAction = () => ({
  type: INITIAL_TEST_ACTION,
})

export type NavigationParams = {
  [string]: any,
}
export type NavigationRoute = NavigationLeafRoute
export type NavigationState = {
  index: number,
  routes: Array<NavigationRoute>,
}

export type NavigationLeafRoute = {|
  key: string,
  routeName: string,
  path?: string,
  params?: NavigationParams,
|}

export type ReactNavigation = NavigationScreenProp

export type GenericObject = {
  [string]: any,
}

export type GenericStringObject = {
  [string]: string,
}

export type MessageAnnotation = {
  name: string,
  version: string,
}

export type TopicAnnotation = {
  tid: number,
  mid: number,
}

export type ReactChildren = Element<*>

export const RESET: 'RESET' = 'RESET'
export const REMOVE_SERIALIZED_CLAIM_OFFERS_SUCCESS: 'REMOVE_SERIALIZED_CLAIM_OFFERS_SUCCESS' =
  'REMOVE_SERIALIZED_CLAIM_OFFERS_SUCCESS'

export type ResetAction = {
  type: typeof RESET,
}

export type ImageSource = {
  uri: string,
}

export type NotificationPayload = {
  forDID: string,
  uid: string,
  type: string,
  remotePairwiseDID: string,
  senderLogoUrl?: ?string,
}

export type ImmutableGenericStringObject = {
  +[string]: string,
}

export const STORE_STATUS = {
  IDLE: 'IDLE',
  IN_PROGRESS: 'IN_PROGRESS',
  ERROR: 'ERROR',
  SUCCESS: 'SUCCESS',
}

export type StoreError = { +error: ?CustomError }
export type StoreStatus = { +status: $Keys<typeof STORE_STATUS> }

export type StatusBarStyle = 'default' | 'light-content' | 'dark-content'
