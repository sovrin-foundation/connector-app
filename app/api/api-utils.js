// @flow
import { SERVER_ERROR_CODE, SERVER_API_CALL_ERROR } from './api-constants'
import { captureError } from '../services/error/error-handler'
import type { CustomError } from '../common/type-common'
import type { ApiData, BackendError } from './type-api'

export const options = (
  method: string = 'GET',
  body: ?{ [string]: any } = null
) => {
  let data: ApiData = {
    method,
    mode: 'cors',
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/json',
    },
  }

  if (body) {
    data.body = JSON.stringify(body)
  }

  return data
}

export const api = (
  url: string,
  apiOptions: ApiData,
  showAlert: boolean = false
) =>
  fetch(url, apiOptions)
    .then(res => {
      // TODO:KS create common method to return successful
      // and unsuccessful api response
      if (res.status >= 200 && res.status < 300) {
        return res
          .json()
          .then(response => ({
            payload: response,
          }))
          .catch((error: Error) => ({
            // our api call was successful,
            // however server did not return any json response
            // Ideally server should send such response with status code 204
            // so that we don't try to parse response for such methods.
            // As of now just send an empty payload signalling that call went fine
            payload: {},
          }))
      } else {
        // Fail with error code if status code is above 300
        return res.text().then((response: string) => ({
          error: response,
        }))
      }
    })
    .then(response => {
      // if response contains payload and no error, that means we got success
      if (response.payload && !response.error) {
        return response.payload
      } else {
        let errorResponse: CustomError = {
          message: response.error,
          code: SERVER_ERROR_CODE,
        }

        try {
          // try to convert error response to json, if it fails that means
          // we did not get error code and message
          const backendError = (JSON.parse(response.error): BackendError)
          errorResponse = {
            message: backendError.statusMsg,
            code: backendError.statusCode,
          }
        } catch (e) {
          // since we did not get error code and message
          // let's just use default that we assigned above,
          // we don't need to do anything here
        }

        throw new Error(JSON.stringify(errorResponse))
      }
    })
    // .catch(captureError) Commenting this for now, it is not returning errors back to saga as of now, Fix this
    .catch((e: Error) => {
      // we don't want to throw JavaScript with simple message
      // we want it to be JSON parse compatible, so that error handlers
      // can parse and use this error
      // Here we are handling network and other http status code errors
      let error = {
        message: e.message,
        code: SERVER_API_CALL_ERROR,
      }

      try {
        // if we already converted JavaScript error to CustomError
        // then pass through same error
        error = JSON.parse(e.message)
      } finally {
      }

      throw new Error(JSON.stringify(error))
    })
