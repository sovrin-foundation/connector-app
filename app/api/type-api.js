// @flow
import type { GenericObject } from '../common/type-common'
import { MESSAGE_TYPE } from './api-constants'
import type { IndyClaimRequest } from '../bridge/react-native-cxs/type-cxs'
import type { ProofApiData } from '../proof-request/type-proof-request'

export type ApiData = {
  method: string,
  mode: string,
  headers: { [string]: string },
  body?: string,
}

export type BackendError = {
  statusCode: string,
  statusMsg: string,
}

export type GetInvitationLinkApiData = {
  agencyUrl: string,
  smsToken: string,
}

export type MessageApiData = {
  agencyUrl: string,
  dataBody: GenericObject,
}

export type DownloadInvitationApiData = {
  url: string,
}

export type SendAuthenticationResponseApiData = {
  data: {
    identifier: string,
    dataBody: GenericObject,
  },
  config: {
    agencyUrl: string,
  },
}

export type SendMessageApiData = {
  agencyUrl: string,
  userPairwiseDid: string,
  agentPayload: GenericObject,
}

export type ApiClaimRequest = {
  userPairwiseDid: string,
  remoteDid: string,
} & IndyClaimRequest

export type SendClaimRequestApiData = {
  agencyUrl: string,
  userPairwiseDid: string,
  responseMsgId: string,
  claimRequest: ApiClaimRequest,
}
export type SendProofApiData = {
  agencyUrl: string,
  userPairwiseDid: string,
  responseMsgId: string,
  proof: ProofApiData,
}

export type EdgeClaimRequest = {
  msg_type: typeof MESSAGE_TYPE.CLAIM_REQUEST,
  version: string,
  to_did: string,
  from_did: string,
  tid: string,
  mid: string,
} & IndyClaimRequest

export type EnvironmentDetailUrlDownloaded = {
  agencyDID: string,
  agencyUrl: string,
  agencyVerificationKey: string,
  poolConfig: string,
}
