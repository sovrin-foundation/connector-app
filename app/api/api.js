// @flow
import { api, options } from './api-utils'
import type {
  GetInvitationLinkApiData,
  MessageApiData,
  DownloadInvitationApiData,
  SendAuthenticationResponseApiData,
  EnvironmentDetailUrlDownloaded,
} from './type-api'
import { PAYLOAD_TYPE, MESSAGE_TYPE, STATUS_CODE } from './api-constants'

export const getInvitationLink = ({
  agencyUrl,
  smsToken,
}: GetInvitationLinkApiData) =>
  api(`${agencyUrl}/agency/url-mapper/${smsToken}`, options('GET'))

export const invitationDetailsRequest = ({ url }: DownloadInvitationApiData) =>
  api(url, options('GET'))

export const getAdditionalData = ({ agencyUrl, dataBody }: MessageApiData) =>
  api(`${agencyUrl}/agency/route`, options('POST', dataBody))

export const sendAuthenticationRequest = ({
  data: { identifier, dataBody },
  config: { agencyUrl },
}: SendAuthenticationResponseApiData) =>
  api(`${agencyUrl}/agent/${identifier}/auth`, options('PUT', dataBody), true)

export const downloadEnvironmentDetails = (
  url: string
): Promise<EnvironmentDetailUrlDownloaded> => api(url, options())
