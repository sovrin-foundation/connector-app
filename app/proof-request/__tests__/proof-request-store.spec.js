// @flow
import { put, call, select } from 'redux-saga/effects'
import { delay } from 'redux-saga'
import proofRequestStore, {
  proofRequestReceived,
  ignoreProofRequest,
  rejectProofRequest,
  acceptProofRequest,
  proofRequestShown,
  sendProof,
  sendProofSuccess,
  sendProofFail,
  proofAccepted,
  convertMissingAttributeListToObject,
  missingAttributesFound,
  proofRequestAutoFill,
  proofRequestShowStart,
} from '../proof-request-store'
import { INITIAL_TEST_ACTION } from '../../common/type-common'
import {
  getProofRequestPairwiseDid,
  getUserPairwiseDid,
  getAgencyUrl,
  getProof,
  getUserOneTimeInfo,
  getAgencyVerificationKey,
  getRemotePairwiseDidAndName,
  getPoolConfig,
} from '../../store/store-selector'

import { MESSAGE_TYPE } from '../../api/api-constants'
import {
  proofRequest,
  proofRequestId as uid,
  poolConfig,
  missingAttributes,
  missingAttributes1,
  fulfilledRequestedAttributes,
} from '../../../__mocks__/static-data'

describe('proof request store', () => {
  const initialAction = { type: 'INITIAL_TEST_ACTION' }
  let initialState = {}
  let newState

  it('should correctly calculate initial state', () => {
    initialState = proofRequestStore(undefined, initialAction)
    expect(initialState).toMatchSnapshot()
  })

  it('proof request is received', () => {
    newState = proofRequestStore(
      initialState,
      proofRequestReceived(proofRequest.payload, proofRequest.payloadInfo)
    )
    expect(newState).toMatchSnapshot()
  })

  it('proof request is shown', () => {
    newState = proofRequestStore(newState, proofRequestShown(uid))
    expect(newState).toMatchSnapshot()
  })

  it('proof request is ignored', () => {
    newState = proofRequestStore(newState, ignoreProofRequest(uid))
    expect(newState).toMatchSnapshot()
  })

  it('proof request is rejected', () => {
    newState = proofRequestStore(newState, rejectProofRequest(uid))
    expect(newState).toMatchSnapshot()
  })

  it('proof request is accepted', () => {
    newState = proofRequestStore(newState, acceptProofRequest(uid))
    expect(newState).toMatchSnapshot()
  })

  it('proof request is ignored', () => {
    newState = proofRequestStore(newState, ignoreProofRequest(uid))
    expect(newState).toMatchSnapshot()
  })

  it('sending proof', () => {
    newState = proofRequestStore(newState, sendProof(uid))
    expect(newState).toMatchSnapshot()
  })

  it('sending proof success', () => {
    newState = proofRequestStore(newState, sendProofSuccess(uid))
    expect(newState).toMatchSnapshot()
  })

  it('sending proof failed', () => {
    newState = proofRequestStore(
      newState,
      sendProofFail(uid, {
        code: 'OCS-002',
        message: 'No pairwise connection found',
      })
    )
    expect(newState).toMatchSnapshot()
  })

  it('ACTION: MISSING_ATTRIBUTES_FOUND', () => {
    expect(
      proofRequestStore(
        newState,
        missingAttributesFound(missingAttributes, uid)
      )
    ).toMatchSnapshot()
  })

  it('ACTION: PROOF_REQUEST_AUTO_FILL', () => {
    expect(
      proofRequestStore(
        newState,
        proofRequestAutoFill(uid, fulfilledRequestedAttributes)
      )
    ).toMatchSnapshot()
  })

  it('should convert missing attributes to self attested attributes', () => {
    expect(
      convertMissingAttributeListToObject(missingAttributes)
    ).toMatchSnapshot()
    expect(
      convertMissingAttributeListToObject(missingAttributes1)
    ).toMatchSnapshot()
  })

  it('should reset proof request store, if RESET action is raised', () => {
    expect(proofRequestStore(newState, { type: 'RESET' })).toMatchSnapshot()
  })

  it('ACTION: PROOF_REQUEST_SHOWN, after missing attributes were found', () => {
    const afterMissingAttributesState = proofRequestStore(
      newState,
      missingAttributesFound(missingAttributes, uid)
    )
    expect(
      proofRequestStore(newState, proofRequestShowStart(uid))
    ).toMatchSnapshot()
  })
})
