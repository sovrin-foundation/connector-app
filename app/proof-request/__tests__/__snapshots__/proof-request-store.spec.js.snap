// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`proof request store ACTION: MISSING_ATTRIBUTES_FOUND 1`] = `
Object {
  "pid123": Object {
    "data": Object {
      "name": "Home Address",
      "requestedAttributes": Array [
        Object {
          "label": "Address 1",
        },
        Object {
          "label": "Address 2",
        },
      ],
      "version": "1.0.0",
    },
    "missingAttributes": Object {
      "sex": Object {
        "data": "",
        "key": "attr2_uuid",
        "name": "sex",
      },
    },
    "originalProofRequestData": Object {
      "name": "proof_req_1",
      "nonce": "123432421212",
      "requested_attributes": Object {
        "attr1_uuid": Object {
          "name": "Address 1",
        },
        "attr2_uuid": Object {
          "name": "Address 2",
        },
      },
      "requested_predicates": Object {},
      "version": "0.1",
    },
    "proofHandle": 1,
    "proofStatus": "SEND_PROOF_FAIL",
    "remotePairwiseDID": "senderDID1",
    "requester": Object {
      "name": "Test Issuer",
    },
    "senderLogoUrl": "http://cr0ybot.github.io/ingress-logos/ingress.png",
    "status": "IGNORED",
    "statusMsg": "pending",
    "uid": "pid123",
  },
}
`;

exports[`proof request store ACTION: PROOF_REQUEST_AUTO_FILL 1`] = `
Object {
  "pid123": Object {
    "data": Object {
      "name": "Home Address",
      "requestedAttributes": Array [
        Object {
          "data": "Address 1",
          "label": "address1",
        },
        Object {
          "data": "Address 2",
          "label": "address2",
        },
      ],
      "version": "1.0.0",
    },
    "originalProofRequestData": Object {
      "name": "proof_req_1",
      "nonce": "123432421212",
      "requested_attributes": Object {
        "attr1_uuid": Object {
          "name": "Address 1",
        },
        "attr2_uuid": Object {
          "name": "Address 2",
        },
      },
      "requested_predicates": Object {},
      "version": "0.1",
    },
    "proofHandle": 1,
    "proofStatus": "SEND_PROOF_FAIL",
    "remotePairwiseDID": "senderDID1",
    "requester": Object {
      "name": "Test Issuer",
    },
    "senderLogoUrl": "http://cr0ybot.github.io/ingress-logos/ingress.png",
    "status": "IGNORED",
    "statusMsg": "pending",
    "uid": "pid123",
  },
}
`;

exports[`proof request store ACTION: PROOF_REQUEST_SHOWN, after missing attributes were found 1`] = `
Object {
  "pid123": Object {
    "data": Object {
      "name": "Home Address",
      "requestedAttributes": Array [
        Object {
          "label": "Address 1",
        },
        Object {
          "label": "Address 2",
        },
      ],
      "version": "1.0.0",
    },
    "missingAttributes": Object {},
    "originalProofRequestData": Object {
      "name": "proof_req_1",
      "nonce": "123432421212",
      "requested_attributes": Object {
        "attr1_uuid": Object {
          "name": "Address 1",
        },
        "attr2_uuid": Object {
          "name": "Address 2",
        },
      },
      "requested_predicates": Object {},
      "version": "0.1",
    },
    "proofHandle": 1,
    "proofStatus": "NONE",
    "remotePairwiseDID": "senderDID1",
    "requester": Object {
      "name": "Test Issuer",
    },
    "senderLogoUrl": "http://cr0ybot.github.io/ingress-logos/ingress.png",
    "status": "RECEIVED",
    "statusMsg": "pending",
    "uid": "pid123",
  },
}
`;

exports[`proof request store proof request is accepted 1`] = `
Object {
  "pid123": Object {
    "data": Object {
      "name": "Home Address",
      "requestedAttributes": Array [
        Object {
          "label": "Address 1",
        },
        Object {
          "label": "Address 2",
        },
      ],
      "version": "1.0.0",
    },
    "originalProofRequestData": Object {
      "name": "proof_req_1",
      "nonce": "123432421212",
      "requested_attributes": Object {
        "attr1_uuid": Object {
          "name": "Address 1",
        },
        "attr2_uuid": Object {
          "name": "Address 2",
        },
      },
      "requested_predicates": Object {},
      "version": "0.1",
    },
    "proofHandle": 1,
    "proofStatus": "NONE",
    "remotePairwiseDID": "senderDID1",
    "requester": Object {
      "name": "Test Issuer",
    },
    "senderLogoUrl": "http://cr0ybot.github.io/ingress-logos/ingress.png",
    "status": "ACCEPTED",
    "statusMsg": "pending",
    "uid": "pid123",
  },
}
`;

exports[`proof request store proof request is ignored 1`] = `
Object {
  "pid123": Object {
    "data": Object {
      "name": "Home Address",
      "requestedAttributes": Array [
        Object {
          "label": "Address 1",
        },
        Object {
          "label": "Address 2",
        },
      ],
      "version": "1.0.0",
    },
    "originalProofRequestData": Object {
      "name": "proof_req_1",
      "nonce": "123432421212",
      "requested_attributes": Object {
        "attr1_uuid": Object {
          "name": "Address 1",
        },
        "attr2_uuid": Object {
          "name": "Address 2",
        },
      },
      "requested_predicates": Object {},
      "version": "0.1",
    },
    "proofHandle": 1,
    "proofStatus": "NONE",
    "remotePairwiseDID": "senderDID1",
    "requester": Object {
      "name": "Test Issuer",
    },
    "senderLogoUrl": "http://cr0ybot.github.io/ingress-logos/ingress.png",
    "status": "IGNORED",
    "statusMsg": "pending",
    "uid": "pid123",
  },
}
`;

exports[`proof request store proof request is ignored 2`] = `
Object {
  "pid123": Object {
    "data": Object {
      "name": "Home Address",
      "requestedAttributes": Array [
        Object {
          "label": "Address 1",
        },
        Object {
          "label": "Address 2",
        },
      ],
      "version": "1.0.0",
    },
    "originalProofRequestData": Object {
      "name": "proof_req_1",
      "nonce": "123432421212",
      "requested_attributes": Object {
        "attr1_uuid": Object {
          "name": "Address 1",
        },
        "attr2_uuid": Object {
          "name": "Address 2",
        },
      },
      "requested_predicates": Object {},
      "version": "0.1",
    },
    "proofHandle": 1,
    "proofStatus": "NONE",
    "remotePairwiseDID": "senderDID1",
    "requester": Object {
      "name": "Test Issuer",
    },
    "senderLogoUrl": "http://cr0ybot.github.io/ingress-logos/ingress.png",
    "status": "IGNORED",
    "statusMsg": "pending",
    "uid": "pid123",
  },
}
`;

exports[`proof request store proof request is received 1`] = `
Object {
  "pid123": Object {
    "data": Object {
      "name": "Home Address",
      "requestedAttributes": Array [
        Object {
          "label": "Address 1",
        },
        Object {
          "label": "Address 2",
        },
      ],
      "version": "1.0.0",
    },
    "originalProofRequestData": Object {
      "name": "proof_req_1",
      "nonce": "123432421212",
      "requested_attributes": Object {
        "attr1_uuid": Object {
          "name": "Address 1",
        },
        "attr2_uuid": Object {
          "name": "Address 2",
        },
      },
      "requested_predicates": Object {},
      "version": "0.1",
    },
    "proofHandle": 1,
    "proofStatus": "NONE",
    "remotePairwiseDID": "senderDID1",
    "requester": Object {
      "name": "Test Issuer",
    },
    "senderLogoUrl": "http://cr0ybot.github.io/ingress-logos/ingress.png",
    "status": "RECEIVED",
    "statusMsg": "pending",
    "uid": "pid123",
  },
}
`;

exports[`proof request store proof request is rejected 1`] = `
Object {
  "pid123": Object {
    "data": Object {
      "name": "Home Address",
      "requestedAttributes": Array [
        Object {
          "label": "Address 1",
        },
        Object {
          "label": "Address 2",
        },
      ],
      "version": "1.0.0",
    },
    "originalProofRequestData": Object {
      "name": "proof_req_1",
      "nonce": "123432421212",
      "requested_attributes": Object {
        "attr1_uuid": Object {
          "name": "Address 1",
        },
        "attr2_uuid": Object {
          "name": "Address 2",
        },
      },
      "requested_predicates": Object {},
      "version": "0.1",
    },
    "proofHandle": 1,
    "proofStatus": "NONE",
    "remotePairwiseDID": "senderDID1",
    "requester": Object {
      "name": "Test Issuer",
    },
    "senderLogoUrl": "http://cr0ybot.github.io/ingress-logos/ingress.png",
    "status": "REJECTED",
    "statusMsg": "pending",
    "uid": "pid123",
  },
}
`;

exports[`proof request store proof request is shown 1`] = `
Object {
  "pid123": Object {
    "data": Object {
      "name": "Home Address",
      "requestedAttributes": Array [
        Object {
          "label": "Address 1",
        },
        Object {
          "label": "Address 2",
        },
      ],
      "version": "1.0.0",
    },
    "originalProofRequestData": Object {
      "name": "proof_req_1",
      "nonce": "123432421212",
      "requested_attributes": Object {
        "attr1_uuid": Object {
          "name": "Address 1",
        },
        "attr2_uuid": Object {
          "name": "Address 2",
        },
      },
      "requested_predicates": Object {},
      "version": "0.1",
    },
    "proofHandle": 1,
    "proofStatus": "NONE",
    "remotePairwiseDID": "senderDID1",
    "requester": Object {
      "name": "Test Issuer",
    },
    "senderLogoUrl": "http://cr0ybot.github.io/ingress-logos/ingress.png",
    "status": "SHOWN",
    "statusMsg": "pending",
    "uid": "pid123",
  },
}
`;

exports[`proof request store sending proof 1`] = `
Object {
  "pid123": Object {
    "data": Object {
      "name": "Home Address",
      "requestedAttributes": Array [
        Object {
          "label": "Address 1",
        },
        Object {
          "label": "Address 2",
        },
      ],
      "version": "1.0.0",
    },
    "originalProofRequestData": Object {
      "name": "proof_req_1",
      "nonce": "123432421212",
      "requested_attributes": Object {
        "attr1_uuid": Object {
          "name": "Address 1",
        },
        "attr2_uuid": Object {
          "name": "Address 2",
        },
      },
      "requested_predicates": Object {},
      "version": "0.1",
    },
    "proofHandle": 1,
    "proofStatus": "SENDING_PROOF",
    "remotePairwiseDID": "senderDID1",
    "requester": Object {
      "name": "Test Issuer",
    },
    "senderLogoUrl": "http://cr0ybot.github.io/ingress-logos/ingress.png",
    "status": "IGNORED",
    "statusMsg": "pending",
    "uid": "pid123",
  },
}
`;

exports[`proof request store sending proof failed 1`] = `
Object {
  "pid123": Object {
    "data": Object {
      "name": "Home Address",
      "requestedAttributes": Array [
        Object {
          "label": "Address 1",
        },
        Object {
          "label": "Address 2",
        },
      ],
      "version": "1.0.0",
    },
    "originalProofRequestData": Object {
      "name": "proof_req_1",
      "nonce": "123432421212",
      "requested_attributes": Object {
        "attr1_uuid": Object {
          "name": "Address 1",
        },
        "attr2_uuid": Object {
          "name": "Address 2",
        },
      },
      "requested_predicates": Object {},
      "version": "0.1",
    },
    "proofHandle": 1,
    "proofStatus": "SEND_PROOF_FAIL",
    "remotePairwiseDID": "senderDID1",
    "requester": Object {
      "name": "Test Issuer",
    },
    "senderLogoUrl": "http://cr0ybot.github.io/ingress-logos/ingress.png",
    "status": "IGNORED",
    "statusMsg": "pending",
    "uid": "pid123",
  },
}
`;

exports[`proof request store sending proof success 1`] = `
Object {
  "pid123": Object {
    "data": Object {
      "name": "Home Address",
      "requestedAttributes": Array [
        Object {
          "label": "Address 1",
        },
        Object {
          "label": "Address 2",
        },
      ],
      "version": "1.0.0",
    },
    "originalProofRequestData": Object {
      "name": "proof_req_1",
      "nonce": "123432421212",
      "requested_attributes": Object {
        "attr1_uuid": Object {
          "name": "Address 1",
        },
        "attr2_uuid": Object {
          "name": "Address 2",
        },
      },
      "requested_predicates": Object {},
      "version": "0.1",
    },
    "proofHandle": 1,
    "proofStatus": "SEND_PROOF_SUCCESS",
    "remotePairwiseDID": "senderDID1",
    "requester": Object {
      "name": "Test Issuer",
    },
    "senderLogoUrl": "http://cr0ybot.github.io/ingress-logos/ingress.png",
    "status": "IGNORED",
    "statusMsg": "pending",
    "uid": "pid123",
  },
}
`;

exports[`proof request store should convert missing attributes to self attested attributes 1`] = `
Object {
  "sex": Object {
    "data": "",
    "key": "attr2_uuid",
    "name": "sex",
  },
}
`;

exports[`proof request store should convert missing attributes to self attested attributes 2`] = `
Object {
  "height": Object {
    "data": "",
    "key": "attr3_uuid",
    "name": "height",
  },
  "sex": Object {
    "data": "",
    "key": "attr2_uuid",
    "name": "sex",
  },
}
`;

exports[`proof request store should correctly calculate initial state 1`] = `Object {}`;

exports[`proof request store should reset proof request store, if RESET action is raised 1`] = `Object {}`;
